lib/src/importer-registry.ts

```diff
@@ -43,10 +53,31 @@ export class ImporterRegistry<sync extends 'sync' | 'async'> {
 
   /** Converts an importer to a proto without adding it to `this.importers`. */
   register(
-    importer: Importer<sync> | FileImporter<sync>
+    importer: Importer<sync> | FileImporter<sync> | NodePackageImporter,
+    _entryPointURL?: string | URL
   ): proto.InboundMessage_CompileRequest_Importer {
     const message = new proto.InboundMessage_CompileRequest_Importer();
-    if ('canonicalize' in importer) {
+    if (typeof importer === 'symbol') {
+      if (importer !== nodePackageImporter) {
+        throw 'Incorrect Node Package Importer used';
+      }
+      const importerMessage = new proto.NodePackageImporter();
+      let entryPointURL = _entryPointURL ?? require.main?.filename;
+      entryPointURL = entryPointURL?.toString();
+      if (entryPointURL === legacyImporterProtocol)
+        entryPointURL = require.main?.filename;
```

Comment:
Multiline `if` statements always need brackets.

Reply:
Addressed in b3f5414d3c366bed91a220089b2f201cec0c178a